lib/vector-core/src/event/util/log/all_fields.rs

```diff
@@ -131,8 +131,14 @@ impl<'a> Iterator for FieldsIter<'a> {
                         }
                     }
                 },
-                Some(LeafIter::Root(value)) => {
-                    return Some(("message".to_owned(), value));
+                Some(LeafIter::Root((value, visited))) => {
```

Comment:
I think you should be able to use `&visited` here to drop the `*` below, and maybe `&value` too (depends on ownership).

Reply:
- `&value` dereferences _twice_, so that doesn't really clean things up (since I want to return a reference)
- `&visited` doesn't compile at all (and would prevent me from mutating the value)

Reply:
Yeah, my bad, I missed that you want to change the value within the stack value.